django/db/models/base.py

```diff
@@ -742,9 +743,9 @@ def refresh_from_db(self, using=None, fields=None, from_queryset=None):
                     field.delete_cached_value(self)
 
         # Clear cached relations.
-        for field in self._meta.related_objects:
-            if (fields is None or field.name in fields) and field.is_cached(self):
-                field.delete_cached_value(self)
+        for rel in self._meta.related_objects:
```

Comment:
`related_objects` doesn't contain field objects, but rel objects, so updating the name to match.

Reply:
We try to avoid refactors/unrelated changes in back ports but as this is very small, I think it's fine :+1: